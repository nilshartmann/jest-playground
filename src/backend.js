const BACKEND_URL = 'http://echo.jsontest.com/greeting/HelloWorld!';

export const loadFromServer = onSuccess => {
    return fetch(BACKEND_URL)
        .then(response => response.json())
        .then(json => onSuccess(json.greeting))
        .catch(err => console.error('SERVER REQUEST FAILED: ' + err))
    ;
};