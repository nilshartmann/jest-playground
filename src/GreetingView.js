import React from 'react';

import {loadFromServer} from './backend';

export default class GreetingView extends React.Component {

    render() {
        if (!this.state) {
            return <p>No Greeting yet</p>;
        }

        return <h1>{this.state.greeting}</h1>;
    }

    componentDidMount() {
        loadFromServer(greeting => this.setState({greeting}))
    }
}