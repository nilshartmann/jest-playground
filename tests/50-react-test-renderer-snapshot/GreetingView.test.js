import React from 'react';
import GreetingView from '../../src/GreetingView';
import renderer from 'react-test-renderer';

import backend from '../../src/backend';

// Setup the module mock. Note that we configure
// the actual mock behaviour of 'loadFromServer'
// in our individual test cases
jest.mock('../../src/backend', () => ({
    loadFromServer: jest.fn()
}));

// Make sure to reset the mocks after each test
beforeEach(() => {
    jest.resetAllMocks();
});

test('it should render fine without greetings', () => {
    const tree = renderer.create(
        <GreetingView />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it should render fine with a message received from backend', () => {
    // Configure the mock. Make it return a fixed answer,
    // that should be rendered by the GreetingView
    backend.loadFromServer.mockImplementation(success => success('moin'));

    const tree = renderer.create(
        <GreetingView />
    ).toJSON();

    // the react-test-renderer runs the componentDidMount hook
    // And as backend.loadFromServer is a sync call (on our mock)
    // componentDidMount have been run completely at this point, so the
    // snapshot should contain the greeting
    expect(tree).toMatchSnapshot();
});
