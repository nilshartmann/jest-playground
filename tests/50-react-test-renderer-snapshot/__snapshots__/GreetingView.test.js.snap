// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render fine with a message received from backend 1`] = `
<h1>
  moin
</h1>
`;

exports[`it should render fine without greetings 1`] = `
<p>
  No Greeting yet
</p>
`;
